### PR TITLE
Fix warnings and enable warningsAsError

### PR DESCRIPTION
**Pull request by ericstj**
This cleans up all warnings in the repo and enables warningAsError so new warnings are not introduced.

### CODE REVIEW

File: src/Microsoft.ML.SearchSpace/Parameter.cs [hunk 1]
```diff
@@ -430,16 +430,28 @@ public override int GetHashCode()
...
+        /// <param name="x">The first parameter to compare.</param>
+        /// <param name="y">The second parameter to compare.</param>
+        /// <returns>true if the value of <paramref name="x" /> is the same as the value of <paramref name="y" />; otherwise, false.</returns>
         public bool Equals(Parameter x, Parameter y)
         {
             return x.GetHashCode() == y.GetHashCode();
         }
 
+        /// <summary>
+        /// <inheritdoc/>
```
Comment (tarekgh) [inline src/Microsoft.ML.SearchSpace/Parameter.cs:445]: >inheritdoc [](http://example.com/codeflow?start=13&length=10)

should this be outside the summary tag? I think we need to fix all occurrence in this file.

File: src/Microsoft.ML.SearchSpace/SearchSpace.cs [hunk 2]
```diff
@@ -155,6 +155,9 @@ public override double[] MappingToFeatureSpace(Parameter parameter)
             return res.ToArray();
         }
 
+        /// <summary>
+        /// <inheritdoc/>
```
Comment (tarekgh) [inline src/Microsoft.ML.SearchSpace/SearchSpace.cs:159]: please have this outside the summary tag.

**Review by tarekgh — APPROVED** [review]

File: src/Microsoft.ML.SearchSpace/Parameter.cs (hunk 1 above)
Comment (ericstj) [reply src/Microsoft.ML.SearchSpace/Parameter.cs:445]: I was just copying all other instances from the same file.  I'll follow up with your suggestions.

**Comment by ericstj** [comment]
@tarekgh I fixed the `<inheritdoc/>` syntax.

**Review by tarekgh — APPROVED** [review]

**Pull request merged**